src/target/riscv/riscv-013.c

```diff
@@ -1787,7 +1789,7 @@ static void riscv013_halt_current_hart(struct target *target)
 
 	/* Issue the halt command, and then wait for the current hart to halt. */
 	uint32_t dmcontrol = dmi_read(target, DMI_DMCONTROL);
-	dmcontrol = set_field(dmcontrol, DMI_DMCONTROL_HALTREQ, 1);
+	dmcontrol |= DMI_DMCONTROL_HALTREQ;
```

Comment:
Is there something wrong with set_field?  IIRC we use it all over the place, if it's broken the we should fix it.  As far as I can tell, these do the same thing so I'm fine with either way.

Reply:
The issue I had when I wrote this code was something about overflow/sign extension, because I think haltreq is the MSB. This may have been fixed elsewhere when we cleaned up the rest of the code, so I can probably try reverting this change.

Reply:
Eh, it's all the same.  As long as set_field works properly then it's not a big deal.